src/keys/get-management-token.ts

```diff
@@ -67,25 +70,45 @@ const getTokenFromOneTimeToken = async (
     `Successfully retrieved CMA Token for app ${appInstallationId} in space ${spaceId} and environment ${environmentId}`
   )
 
-  return JSON.parse(response.body).token
+  return JSON.parse(response.body).token as Promise<string>
 }
 
 /**
  * Factory method for GetManagementToken
  * @internal
  */
 export const createGetManagementToken = (log: Logger, http: HttpClient) => {
-  return (privateKey: unknown, opts: GetManagementTokenOptions): Promise<string> => {
+  return async (privateKey: unknown, opts: GetManagementTokenOptions): Promise<string> => {
     if (!(typeof privateKey === 'string')) {
       throw new ReferenceError('Invalid privateKey: expected a string representing a private key')
     }
 
+    const cacheKey = opts.appInstallationId + opts.environmentId + privateKey
```

Comment:
Private keys can get loooong. Shall we maybe truncate it to to first `100` characters to same some memory overhead?

Reply:
I considered hashing it but didn't want the overhead. Truncating is a good option, I'll do that. Also closes potential issues with node-cache exposing data or something along those lines.